bootloader/grub.go

```diff
@@ -440,6 +455,39 @@ func (g *grub) CandidateCommandLine(pieces CommandLineComponents) (string, error
 	return g.commandLineForEdition(edition, pieces)
 }
 
+// DefaultCommandLine returns the default kernel command-line used by
+// the bootloader excluding the recovery mode and system parameters.
+func (g *grub) DefaultCommandLine(candidate bool) (string, error) {
+	var edition uint
+
+	// if "candidate", we look for the managed boot assets
+	// (current snapd) rather than the ones currently installed on
+	// the boot/seed disk. This is needed to know the default
+	// command line before candidate boot assets are installed
+	if candidate {
+		assetName := "grub.cfg"
+		if g.recovery {
+			assetName = "grub-recovery.cfg"
+		}
```

Comment:
I've seen this twice now, maybe nice to move it to a `func (g *grub) assetName() string` function to avoid hardcoding these names multiple times?

Reply:
Done. There is a place in the file where we take an option as argument instead of looking at the "intention" from the grub struct. So it cannot really be used there.